src/GitHub.TeamFoundation.16/Services/VsDiffBase.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using GitHub.Logging;
+using GitHub.Services;
+using Microsoft.VisualStudio.RichReview.Contracts;
+using Microsoft.VisualStudio.RichReview.UX.Controls;
+using Microsoft.VisualStudio.RichReview.UX.Controls.ViewModel;
+using Microsoft.VisualStudio.Shell;
+using Serilog;
+
+namespace GitHub.TeamFoundation.Services
+{
+    internal class VsDiffBaseFactory
+    {
+        [ImportingConstructor]
+        public VsDiffBaseFactory(
+            [Import("Microsoft.VisualStudio.RichReview.Contracts.IDiffBaseService", AllowDefault = true)] Lazy<object> diffBaseServiceObject)
+        {
+            if (diffBaseServiceObject != null)
+            {
+                Instance = new VsDiffBase(diffBaseServiceObject);
+            }
+        }
+
+        [Export(typeof(IVsDiffBase))]
+        internal IVsDiffBase Instance { get; }
+    }
+
+    internal class VsDiffBase : IVsDiffBase
+    {
+        static readonly ILogger log = LogManager.ForContext<VsDiffBase>();
+
+        readonly Lazy<object> diffBaseServiceObject;
+
+        // store this in a static because this class does not seem to be a singleton
+        static object ChangesList;
```

Comment:
I'm surprised about this. I thought MEF components were singletons by default? 😕